csharp/src/Apache.Arrow.Flight.Sql/Client/FlightSqlClient.cs

```diff
@@ -0,0 +1,1211 @@
+        }
+        catch (RpcException ex)
+        {
+            throw new InvalidOperationException("Failed to cancel flight info", ex);
+        }
+    }
+
+    /// <summary>
+    /// Explicitly cancel a query.
+    /// </summary>
+    /// <param name="options">RPC-layer hints for this call.</param>
+    /// <param name="info">The FlightInfo of the query to cancel.</param>
+    /// <returns>A Task representing the asynchronous operation.</returns>
+    public async Task<CancelStatus> CancelQueryAsync(FlightCallOptions options, FlightInfo info)
+    {
+        if (options == null)
+        {
+            throw new ArgumentNullException(nameof(options));
+        }
+
+        if (info == null)
+        {
+            throw new ArgumentNullException(nameof(info));
+        }
+
+        try
+        {
+            var cancelRequest = new CancelFlightInfoRequest(info);
+            var action = new FlightAction("CancelFlightInfo", cancelRequest.ToByteString());
+            var call = _client.DoAction(action, options.Headers);
+            await foreach (var result in call.ResponseStream.ReadAllAsync())
+            {
+                var cancelResult = Any.Parser.ParseFrom(result.Body);
+                if (cancelResult.TryUnpack(out CancelFlightInfoResult cancelFlightInfoResult))
+                {
+                    return cancelFlightInfoResult.CancelStatus switch
+                    {
+                        CancelStatus.Cancelled => CancelStatus.Cancelled,
+                        CancelStatus.Cancelling => CancelStatus.Cancelling,
+                        CancelStatus.NotCancellable => CancelStatus.NotCancellable,
+                        _ => CancelStatus.Unspecified
+                    };
+                }
+            }
+
+            throw new InvalidOperationException("Failed to cancel query: No response received.");
+        }
+        catch (RpcException ex)
+        {
+            throw new InvalidOperationException("Failed to cancel query", ex);
+        }
+    }
+
+    /// <summary>
+    /// Begin a new transaction.
+    /// </summary>
+    /// <param name="options">RPC-layer hints for this call.</param>
+    /// <returns>A Task representing the asynchronous operation. The task result contains the Transaction object representing the new transaction.</returns>
+    public async Task<Transaction> BeginTransactionAsync(FlightCallOptions options)
+    {
+        if (options == null)
+        {
+            throw new ArgumentNullException(nameof(options));
+        }
+
+        try
+        {
+            var actionBeginTransaction = new ActionBeginTransactionRequest();
+            var action = new FlightAction("BeginTransaction", actionBeginTransaction.PackAndSerialize());
+            var responseStream = _client.DoAction(action, options.Headers);
+            await foreach (var result in responseStream.ResponseStream.ReadAllAsync())
```

Comment:
Add `ConfigureAwait(false)` if supported (I've not tried on a foreach)

Reply:
done